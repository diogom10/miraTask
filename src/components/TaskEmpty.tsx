import React from 'react';
import {Task} from '../models/Task';
import {View, Text, StyleSheet, TouchableOpacity} from 'react-native';
import {Warning} from '../assets/icons/warning';

interface ITaskEmptyProps {
  title: string;
  actionTitle?: string;
  action?: () => void;
}

export const TaskEmpty = ({
  title,
  actionTitle,
  action = () => {},
}: ITaskEmptyProps) => (
  <View style={styles.container}>
    <Warning style={styles.icon} width={80} height={80} fill={'#656053'} />
    <Text style={styles.title}>{title}</Text>
    <TouchableOpacity onPress={action}>
      <Text style={styles.actionTitle}>{actionTitle}</Text>
    </TouchableOpacity>
  </View>
);

const styles = StyleSheet.create({
  container: {
    borderColor: '#fff',
    borderRadius: 20,
    minHeight: 200,
    borderWidth: 1,
    padding: 4,
    margin: 2,
    marginTop: '50%',
    paddingHorizontal: 15,
    paddingVertical: 15,
    backgroundColor: '#fbfbfb',
    marginBottom: 10,
    alignItems: 'center',
  },
  title: {
    fontSize: 22,
    fontWeight: 'bold',
    textAlign: 'center',
    width: '92%',
  },
  actionTitle: {
    fontSize: 14,
    fontWeight: 'bold',
    textDecorationLine: 'underline',
    textDecorationColor: '#508ce8',
    color: '#508ce8',
    textAlign: 'center',
    width: '92%',
  },
  icon: {
    marginBottom: 20,
  },
});
