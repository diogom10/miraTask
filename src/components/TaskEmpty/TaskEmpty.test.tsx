import React from 'react';
import {act, create} from 'react-test-renderer';
import {ITaskEmptyProps, TaskEmpty} from './TaskEmpty';

const props: ITaskEmptyProps = {
  title: 'You don`t have any task',
  actionTitle: 'Try add a new task',
  action: jest.fn(),
  testID: 'TaskEmpty',
};

const component = create(<TaskEmpty {...props} />);
const el_title = `${props.testID}_title`;
const el_btn_action = `${props.testID}_btn_action`;
const el_action_title = `${props.testID}_action_title`;

describe('TaskView Component', () => {
  test(`should render label #${el_title}`, async () => {
    const label = component.root.findByProps({testID: el_title}).props;
    expect(label).toBeTruthy();
  });
  test(`should render label #${el_action_title}`, async () => {
    const label = component.root.findByProps({testID: el_action_title}).props;
    expect(label).toBeTruthy();
  });
  test(`should render btn #${el_btn_action}`, async () => {
    const button = component.root.findByProps({testID: el_btn_action}).props;
    act(() => button.onPress());
  });
  test('snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
