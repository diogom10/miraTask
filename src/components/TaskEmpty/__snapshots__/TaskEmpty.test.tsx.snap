// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TaskView Component snapshot 1`] = `
<View
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#fbfbfb",
      "borderColor": "#fff",
      "borderRadius": 20,
      "borderWidth": 1,
      "margin": 2,
      "marginBottom": 10,
      "marginTop": "50%",
      "minHeight": 200,
      "padding": 4,
      "paddingHorizontal": 15,
      "paddingVertical": 15,
    }
  }
>
  <RNSVGSvgView
    align="xMidYMid"
    bbHeight="80"
    bbWidth="80"
    fill="#656053"
    focusable={false}
    height={80}
    meetOrSlice={0}
    minX={0}
    minY={0}
    style={
      [
        {
          "backgroundColor": "transparent",
          "borderWidth": 0,
        },
        {
          "marginBottom": 20,
        },
        {
          "flex": 0,
          "height": 80,
          "width": 80,
        },
      ]
    }
    vbHeight={24}
    vbWidth={24}
    width={80}
  >
    <RNSVGGroup
      fill={
        {
          "payload": 4284833875,
          "type": 0,
        }
      }
      propList={
        [
          "fill",
        ]
      }
    >
      <RNSVGPath
        d="M12 2c5.514 0 10 4.486 10 10s-4.486 10-10 10-10-4.486-10-10 4.486-10 10-10zm0-2c-6.627 0-12 5.373-12 12s5.373 12 12 12 12-5.373 12-12-5.373-12-12-12zm-1 6h2v8h-2v-8zm1 12.25c-.69 0-1.25-.56-1.25-1.25s.56-1.25 1.25-1.25 1.25.56 1.25 1.25-.56 1.25-1.25 1.25z"
        fill={
          {
            "payload": 4284833875,
            "type": 0,
          }
        }
        propList={
          [
            "fill",
          ]
        }
      />
    </RNSVGGroup>
  </RNSVGSvgView>
  <Text
    style={
      {
        "fontSize": 22,
        "fontWeight": "bold",
        "textAlign": "center",
        "width": "92%",
      }
    }
    testID="TaskEmpty_title"
  >
    You don\`t have any task
  </Text>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "opacity": 1,
      }
    }
    testID="TaskEmpty_btn_action"
  >
    <Text
      style={
        {
          "color": "#508ce8",
          "fontSize": 14,
          "fontWeight": "bold",
          "textAlign": "center",
          "textDecorationColor": "#508ce8",
          "textDecorationLine": "underline",
          "width": "92%",
        }
      }
      testID="TaskEmpty_action_title"
    >
      Try add a new task
    </Text>
  </View>
</View>
`;
